rust/lance/src/dataset/cleanup.rs

```diff
@@ -117,13 +120,13 @@ const UNVERIFIED_THRESHOLD_DAYS: i64 = 7;
 impl<'a> CleanupTask<'a> {
     fn new(
         dataset: &'a Dataset,
-        before: DateTime<Utc>,
+        policy: CleanupPolicy,
         delete_unverified: bool,
```

Comment:
why `delete_unverified` and `error_if_old_versions_tagged` are not a part of the policy?

Reply:
My thoughts on this is: the CleanupPolicy only determines which manifests need to be cleaned. The ```delete_unverified``` and ```error_if_old_versions_tagged``` determines the behavior of cleanup task.

```
#[derive(Clone, Debug)]
struct CleanupTask<'a> {
    dataset: &'a Dataset,
    /// The policy determines which versions should be cleanup
    policy: CleanupPolicy,
    /// If true, delete unverified data files even if they are recent
    delete_unverified: bool,
    /// If true, return an Error if a tagged version is old
    error_if_old_versions_tagged: bool,
}
```